agent/xds/listeners.go

```diff
@@ -1121,10 +1154,19 @@ func (s *ResourceGenerator) makeFilterChainTerminatingGateway(
 		return nil, err
 	}
 
-	filterChain := &envoy_listener_v3.FilterChain{
-		FilterChainMatch: makeSNIFilterChainMatch(cluster),
-		Filters:          make([]*envoy_listener_v3.Filter, 0, 3),
-		TransportSocket:  transportSocket,
+	var filterChain *envoy_listener_v3.FilterChain
+	if dest != nil {
+		filterChain = &envoy_listener_v3.FilterChain{
+			FilterChainMatch: makeEndpointFilterChainMatch(cluster, dest),
+			Filters:          make([]*envoy_listener_v3.Filter, 0, 3),
+			TransportSocket:  transportSocket,
```

Comment:
I'm a bit confused about this, what would happen if we have let say a hostname as an Address. 
I think we will end up with duplicate filters, am I mistaken?

Reply:
Do you mean duplicates like the "tcp_proxy" and "sni_dynamic_forward_proxy" filters? [All of the docs ](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/sni_dynamic_forward_proxy_filter)use both filters in tandem and I kept the RBAC one for TLS. Or are you talking about de-duplicating similar hostnames?